### DISCUSSION TITLE
Update / Aktualisierung eScriptorium

### DISCUSSION THREAD

**Discussion by stweil** [UB-Mannheim/escriptorium#8] [Announcements]
Dear all,

currently we run an eScriptorium software version from July/August 2023.

Meanwhile the latest code adds a modern new user interface (UI), but still provides a user specific option for the old "legacy" user interface. It also comes with the latest kraken OCR with faster training. This new code hat some nasty bugs which seem to be fixed now. So maybe it's time to upgrade our installation.

We still have to decide about the time for the upgrade and can take your needs into account. So please comment here or write us directly. As soon as we have planned the upgrade, we will give more information here. Be prepared to make backups of your valuable data with the export function of eScriptorium.

Your
eScriptorium Team at UB Mannheim

Update: The upgrade with the new user interface was installed on 2025-12-23, see details below.

**Comment by johnlockejrr** [comment]
Is there a way to use locally your "build" to test it? I don't mean the official `eScriptorium`.

**Reply by stweil** [reply]
Sure, just install https://github.com/UB-Mannheim/escriptorium/tree/es_next instead of the official develop branch.

**Comment by johnlockejrr** [comment]
How do you run with `docker` on GPU? Seems `nvidia-docker2` is deprecated. Following `https://gitlab.com/scripta/escriptorium/-/wikis/docker-install` doesn't seem to run on GPU. I normally use a full install of `eScriptorium` to run on GPU. The celery image is run on GPU as seen below but the tasks are not run on GPU, very strange. Thanks!

```
570ee13caf16   registry.gitlab.com/scripta/escriptorium:latest   "/usr/src/app/entryp…"   13 hours ago   Up 12 hours                            escriptorium-ub-mannheim-celery-gpu-1
```

```
# ps xa
  PID TTY      STAT   TIME COMMAND
    1 ?        Ssl    2:00 /usr/local/bin/python /usr/local/bin/celery -A escriptorium worker -l INFO -E -Ofair --prefetch-multiplier 1 -Q gpu -c 1 --max-tasks-per-child=1
  258 ?        Zs     0:00 [torch_shm_manag] <defunct>
  259 ?        Zs     0:00 [torch_shm_manag] <defunct>
  260 ?        Zs     0:00 [torch_shm_manag] <defunct>
  261 ?        Zs     0:00 [torch_shm_manag] <defunct>
  279 ?        Zs     0:00 [torch_shm_manag] <defunct>
  280 ?        Zs     0:00 [torch_shm_manag] <defunct>
  281 ?        Zs     0:00 [torch_shm_manag] <defunct>
  282 ?        Zs     0:00 [torch_shm_manag] <defunct>
  318 ?        Z      0:02 [celery] <defunct>
  319 ?        Z      0:02 [celery] <defunct>
  320 ?        Z      0:01 [celery] <defunct>
  321 ?        Z      0:01 [celery] <defunct>
  322 ?        Z      0:02 [celery] <defunct>
  323 ?        Z      0:01 [celery] <defunct>
  324 ?        Z      0:02 [celery] <defunct>
  325 ?        Z      0:02 [celery] <defunct>
  344 ?        S      0:00 /usr/local/bin/python /usr/local/bin/celery -A escriptorium worker -l INFO -E -Ofair --prefetch-multiplier 1 -Q gpu -c 1 --max-tasks-per-child=1
  345 pts/0    Ss     0:00 sh
  353 pts/0    R+     0:00 ps xa
# nvidia-smi
Mon Nov  4 11:51:15 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 565.51.01              Driver Version: 565.90         CUDA Version: 12.7     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3060        On  |   00000000:04:00.0 Off |                  N/A |
|  0%   24C    P8             10W /  170W |       0MiB /  12288MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
```

**Reply by stweil** [reply]
We use native installations, no Docker.

**Comment by johnlockejrr** [comment]
Same here, I just thought I'll give a try to docker for a fast check but I get back to native install because I can controll it better.

**Comment by johnlockejrr** [comment]
What `python` version and `nodejs` version do you recommand?

**Comment by stweil** [comment]
Python 3.11, Nodejs 18.19 (both packages in Debian stable).

**Reply by stweil** [reply]
Meanwhile we run Debian trixie (the current Debian stable) with Nodejs 20.19.2, but still use Python 3.11 (not 3.13 which is part of Debian trixie, see related issue https://github.com/mittagessen/kraken/issues/718).

**Comment by stweil** [comment]
Dear all,

Now, one year later, I think it is time to update our installation.

Several public eScriptorium instances based on recent code have emerged in the meantime, such as https://sofer.info/, https://escriptorium.rich.ru.nl/, https://escriptorium.d4science.org/, https://chedil.huma-num.fr/ and https://www.peritextum.app/.

I intend to update the installation at https://ocr-bw.bib.uni-mannheim.de/escriptorium/ during the Christmas break. Our service won't be available during the update process.

**Comment by stweil** [comment]
The new release is now installed. It is still possible to select the old user interface ("Legacy mode") in the [personal settings](https://ocr-bw.bib.uni-mannheim.de/escriptorium/profile/).

Major changes:
- New user interface.
- Latest kraken release (faster training).
- Please be aware that the epoch versions of the models will only be available for 30 days.

Known issues:
- [Line masks not updating](https://gitlab.com/scripta/escriptorium/-/issues/1177)
- ["Transcribe" without "Segment" does not work with new UI](https://gitlab.com/scripta/escriptorium/-/issues/1178)

**Comment by stweil** [comment]
Meanwhile (2026-01-30) we run [eScriptorium v1.0.0](https://gitlab.com/scripta/escriptorium/-/releases/v1.0.0).
